buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -0,0 +1,79 @@
+package com.google.firebase.gradle
+
+import java.io.File
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.ListBranchCommand
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.lib.Constants
+import org.eclipse.jgit.lib.ObjectId
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+data class FirebaseLibrary(val moduleNames: List<String>, val directories: List<String>)
+
+open class ReleaseGenerator : DefaultTask() {
+    @TaskAction
+    @Throws(Exception::class)
+    fun generateReleaseConfig() {
+        val currentRelease = project.property("currentRelease").toString()
+        val pastRelease = project.property("pastRelease").toString()
+        val rootDir = project.rootDir
+        val availableModules = parseSubProjects(rootDir)
+        val firebaseLibraries = extractLibraries(availableModules)
+
+        val repo = Git.open(rootDir)
+        val headRef = repo.repository.resolve(Constants.HEAD)
+        val branchRef = getObjectRefForBranchName(repo, pastRelease)
+
+        val changedDirs = getChangedLibraries(repo, branchRef, headRef, firebaseLibraries)
+        writeReleaseConfig(rootDir, changedDirs, currentRelease)
```

Comment:
Could you add a param to print the list of libraries as a comma-separated string too? That output could eventually be passed directly to the `firebasePublish` command as a property, i.e. `-PprojectsToPublish=` removing the need for the config file itself. Just to future proof ourselves.